RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotESProducer.cc

```diff
@@ -30,13 +30,22 @@ class OnlineBeamSpotESProducer : public edm::ESProducer {
 
 private:
   const BeamSpotOnlineObjects* compareBS(const BeamSpotOnlineObjects* bs1, const BeamSpotOnlineObjects* bs2);
-  BeamSpotObjects fakeBS_;
+  const BeamSpotOnlineObjects* checkSingleBS(const BeamSpotOnlineObjects* bs1);
 
   edm::ESGetToken<BeamSpotObjects, BeamSpotTransientObjectsRcd> const bsToken_;
   edm::ESGetToken<BeamSpotOnlineObjects, BeamSpotOnlineHLTObjectsRcd> bsHLTToken_;
   edm::ESGetToken<BeamSpotOnlineObjects, BeamSpotOnlineLegacyObjectsRcd> bsLegacyToken_;
+
+  BeamSpotObjects fakeBS_;
+  int timeThreshold_;
+  double sigmaZThreshold_;
 };
+
 OnlineBeamSpotESProducer::OnlineBeamSpotESProducer(const edm::ParameterSet& p) {
+  // get parameters
+  timeThreshold_ = p.getParameter<int>("timeThreshold");
```

Comment:
```suggestion
OnlineBeamSpotESProducer::OnlineBeamSpotESProducer(const edm::ParameterSet& p)
  // get parameters
  timeThreshold_(p.getParameter<int>("timeThreshold")), ...
```
initializer list is preferred to the constructor body and also allows to maintain constness of the parameters